### PR TITLE
Network feature: Setting custom ipsec policy for Virtual Network Gateway P2S clients.

### PR DESCRIPTION
**Pull request by Nilambari**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/stable/2018-01-01/virtualNetworkGateway.json [hunk 1]
```diff
@@ -675,7 +675,98 @@
         "x-ms-long-running-operation": true
       }
     },
-    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/connections/{virtualNetworkGatewayConnectionName}/vpndeviceconfigurationscript": {
+	"/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworkGateways/{virtualNetworkGatewayName}/setvpnclientipsecparameters": {
```
Comment (anuchandy) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-01-01/virtualNetworkGateway.json]: Please fix indentation

File: specification/network/resource-manager/Microsoft.Network/stable/2018-01-01/virtualNetworkGateway.json [hunk 2]
```diff
@@ -675,7 +675,98 @@
...
+            "schema": {
+              "$ref": "#/definitions/VpnClientIPsecParameters"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/virtualNetworkGateways/{virtualNetworkGatewayName}/getvpnclientipsecparameters": {
+      "post": {
```
Comment (anuchandy) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-01-01/virtualNetworkGateway.json]: You mean `GET`? please confirm..

Comment (anuchandy) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-01-01/virtualNetworkGateway.json]: Or is this operation a Long Running GET? if so annotate it accordingly

File: specification/network/resource-manager/Microsoft.Network/stable/2018-01-01/virtualNetworkGateway.json [hunk 3]
```diff
@@ -2140,15 +2242,17 @@
...
           "enum": [
             "None",
             "PFS1",
             "PFS2",
             "PFS2048",
             "ECP256",
             "ECP384",
-            "PFS24"
+            "PFS24",
+			"PFS14",
```
Comment (anuchandy) [inline specification/network/resource-manager/Microsoft.Network/stable/2018-01-01/virtualNetworkGateway.json]: Fix the indentation

**Review by anuchandy — CHANGES_REQUESTED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2018-01-01/virtualNetworkGateway.json (hunk 2 above)
Comment (Nilambari) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-01-01/virtualNetworkGateway.json]: Yes, it’s a LRO POST. Annotated accordingly.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-01-01/virtualNetworkGateway.json (hunk 1 above)
Comment (Nilambari) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-01-01/virtualNetworkGateway.json]: Done.

File: specification/network/resource-manager/Microsoft.Network/stable/2018-01-01/virtualNetworkGateway.json (hunk 3 above)
Comment (Nilambari) [reply specification/network/resource-manager/Microsoft.Network/stable/2018-01-01/virtualNetworkGateway.json]: Done.

**Comment by Nilambari** [comment]
Just talked to Deepak, looks like I need to wait for swagger changes till NRP changes do not go live so I will cancel current PR: https://github.com/Azure/azure-rest-api-specs/pull/2341  and will reopen once NRP deployment starts.

**Pull request closed without merging**